lib/core/options/builders.js

```diff
@@ -0,0 +1,483 @@
+				return new RegExp(`^(${data.patt})$`, 'i').test(sub);
+			},
+		},
+		currentUserProfile: {
+			name: 'When browsing a user profile',
+			defaultTemplate(patt) {
+				return { type: 'currentUserProfile', patt: patt || '' };
+			},
+			fields: [
+				'when browsing /u/',
+				{ type: 'text', id: 'patt', validator: RegExp },
+				'\'s posts',
+			],
+			evaluate(thing, data) {
+				const user = currentUserProfile();
+				if (user === null) return false;
+				return new RegExp(`^(${data.patt})$`, 'i').test(user);
+			},
+		},
+		currentMulti: {
+			name: 'When browsing a multireddit',
+			defaultTemplate(user, name) {
+				return { type: 'currentMulti', user: user || '', name: name || '' };
+			},
+			fields: [
+				'when browsing /u/',
+				{ type: 'text', id: 'user', validator: RegExp },
+				'/m/',
+				{ type: 'text', id: 'name', validator: RegExp },
+			],
+			evaluate(thing, data) {
+				const parts = (/^(?:user\/)?([a-z0-9_-]+)\/m\/([a-z0-9_-]+)$/i)
+					.exec(currentMultireddit());
+				if (parts === null) return false;
+				const multiNameRE = data.name.trim() === '' ? /.*/ : new RegExp(`^(${data.name})$`, 'i');
+				const user = parts[1];
+				const multi = parts[2];
+				if (user === 'me' && data.name.trim() === 'me') {
+					return multiNameRE.test(multi);
+				} else {
+					return (data.user.trim() === '' ? /.*/ : new RegExp(`^(${data.user})$`, 'i')).test(user) && multiNameRE.test(multi);
+				}
+			},
+		},
+		browsingFrontPage: {
+			name: 'Browsing the front page',
+			defaultTemplate() {
+				return { type: 'browsingFrontPage' };
+			},
+			fields: [
+				'when browsing the front page',
+			],
+			evaluate() {
+				return isPageType('linklist') &&
+					currentSubreddit() === null &&
+					currentMultireddit() === null &&
+					currentUserProfile() === null;
+			},
+		},
+		isNSFW: {
+			name: 'NSFW post',
+			defaultTemplate() {
+				return { type: 'isNSFW' };
+			},
+			fields: [
+				'post is marked NSFW',
+			],
+			evaluate(thing) {
+				return thing.classList.contains('over18');
+			},
+		},
+		loggedInAs: {
+			name: 'Logged in user',
+			defaultTemplate(patt) {
+				return { type: 'loggedInAs', patt: patt || '' };
+			},
+			fields: [
+				'logged in as /u/',
+				{ type: 'text', id: 'loggedInAs', validator: RegExp },
+			],
+			evaluate(thing, data) {
+				const myName = loggedInUser();
+				if (!myName) {
+					return false;
+				}
+				return new RegExp(`^(${data.loggedInAs})$`, 'i').test(myName);
+			},
+		},
+		toggle: {
+			name: 'Custom toggle',
+			defaultTemplate() {
+				return { type: 'toggle', toggleName: '' };
+			},
+			fields: [
+				'custom toggle named',
+				{ type: 'text', id: 'toggleName', validator: RegExp },
+				'is turned on',
+			],
+			evaluate(thing, data) {
+				const toggleName = data.toggleName;
+				return CustomToggles.toggleActive(toggleName);
+			},
+		},
+		/*
+		possible for later:
+
+		res tag
+		res score
+		up/down/unvoted
+		sort
+		//*/
+	},
+	... options });
```

Comment:
Nit: newline before `});`, no space around `...` (both will be linted soon<sup>TM</sup>)